core/src/lib.rs

```diff
@@ -45,7 +49,7 @@ pub mod types;
 pub type Result<T> = std::result::Result<T, Error>;
 
 /// A `Future` trait object.
-pub type BoxFuture<T> = Pin<Box<dyn futures::Future<Output = T> + Send>>;
+pub type BoxFuture<T> = Pin<Box<dyn std::future::Future<Output = T> + Send>>;
```

Comment:
any reason not to use https://docs.rs/futures/0.3.13/futures/future/type.BoxFuture.html instead of this?